### DISCUSSION TITLE
XMLExporter?

### DISCUSSION THREAD

**Discussion by tychobrailleur** [ho-dev/HattrickOrganizer#1942] [General]
As I am continuing my efforts to port code to Kotlin, I am currently looking at `XMLExporter` (invoked from Tools > XML Exporter), whose intent seems to be to re-export the list of matches of the HO DB into an XML file.

I am currently scratching my head over this class, as:
- Currently, it doesn't seem to return any of the matches in my DB, mostly because of the `isValid` checks, where it cannot find matching HRF IDs for most entries.  Excluding that logic, there is all sorts of checks that don't quite make sense either to exclude the match (has red card??)
- I just don't get the point of that export.  The format doesn't seem to match CHPP, so to use this you'd need to have your own parsing logic, and as far as I am aware, I can't see any logic to re-import this file.

Rather porting this blindly, I was thinking to ask:
- Is anyone familiar with this feature, and what it is exactly supposed to do?
- Should we not just remove it, and instead write an export that would be a bit more useful (in XML or JSON), in particular making it re-importable into HO?

**Comment by wsbrenk** [comment]
@tychobrailleur I'm not familiar with this feature and agree your suggestion to remove it. 
the matches can be downloaded from HT again - so i don't see any reason to export and import it by files.

maybe this feature is/was used to feed other tools (maybe used by akasolace when analyzing his kind of rating prediction?)

**Reply by tychobrailleur** [reply]
Sounds good, will do that.
